app/controllers/admins_controller.rb

```diff
@@ -0,0 +1,53 @@
+class AdminsController < ApplicationController
+  before_action :set_admin, only: [:show, :update, :destroy]
+
+  # GET /admins
+  # GET /admins.json
+  def index
+    @admins = Admin.all
+  end
+
+  # GET /admins/1
+  # GET /admins/1.json
+  def show
+  end
+
+  # POST /admins
+  # POST /admins.json
+  def create
+    @admin = Admin.new(admin_params)
+
+    if @admin.save
+      render :show, status: :created, location: @admin
+    else
+      render json: @admin.errors, status: :unprocessable_entity
+    end
+  end
+
+  # PATCH/PUT /admins/1
+  # PATCH/PUT /admins/1.json
+  def update
+    if @admin.update(admin_params)
+      render :show, status: :ok, location: @admin
+    else
+      render json: @admin.errors, status: :unprocessable_entity
+    end
+  end
+
+  # DELETE /admins/1
+  # DELETE /admins/1.json
+  def destroy
+    @admin.destroy
+  end
+
+  private
+    # Use callbacks to share common setup or constraints between actions.
+    def set_admin
+      @admin = Admin.find(params[:id])
```

Comment:
`find(123)` will crash if admin with id `123` does not exist, you can prevent this by using `find_by(id: params[:id])` which will return `nil`  if admin does not exist, then you can handle the `nil` value in your action.

> User.find(id: 123)  # raises ActiveRecord::RecordNotFound when no record is found


> User.find_by(id: 123) # returns nil if no record found

Just so you know next time 👍